### PR TITLE
Markdown editor

### PR DESCRIPTION
**Pull request by Prospector**

### CODE REVIEW

File: lib/components/base/MarkdownEditor.vue [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+})
+
+const videoUrl = ref('')
+const youtubeRegex =
+  /^(?:https?:)?(?:\/\/)?(?:youtu\.be\/|(?:www\.|m\.)?youtube\.com\/(?:watch|v|embed)(?:\.php)?(?:\?.*v=|\/))([a-zA-Z0-9_-]{7,15})(?:[?&][a-zA-Z0-9_-]+=[a-zA-Z0-9_-]+)*$/
+
+const videoMarkdown = computed(() => {
+  const match = youtubeRegex.exec(videoUrl.value)
+  if (match) {
+    return `<iframe width="560" height="315" src="https://www.youtube.com/embed/${match[1]}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
```
Comment (triphora) [inline lib/components/base/MarkdownEditor.vue:292]: Probably better to have it in nocookie mode [like so](https://www.youtube-nocookie.com/embed/d1YBv2mWll0)

```suggestion
    return `<iframe width="560" height="315" src="https://www.youtube-nocookie.com/embed/${match[1]}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
```

File: docs/components/markdown-editor.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+```vue
+<script setup>
+import { ref } from "vue";
+const description = ref(null)
+</script>
+
+<MarkdownEditor v-model="description" :heading-buttons="false" />
+```
+a
```
Comment (Erb3) [inline docs/components/markdown-editor.md:37]: a?

Comment (triphora) [reply docs/components/markdown-editor.md:37]: a

Comment (osfanbuff63) [reply docs/components/markdown-editor.md:37]: a!

Comment (kingmaj0r) [reply docs/components/markdown-editor.md:37]: a

**Review by Geometrically — APPROVED** [review]

**Pull request merged**